### PR TITLE
GG-14830

### PR DESCRIPTION
**Pull request by SammyVimes**
- Add test for checking that temporary checkpoint files are getting deleted on startup

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        cfg.setConsistentId(igniteInstanceName);
+
+        DataStorageConfiguration storageCfg = new DataStorageConfiguration();
+
+        storageCfg.getDefaultDataRegionConfiguration()
+            .setPersistenceEnabled(true);
+
+        cfg.setDataStorageConfiguration(storageCfg);
+
+        cfg.setFailureHandler(new StopNodeFailureHandler());
```
Comment (alievmirza) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:67]: Do you really need this setting?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import org.apache.ignite.events.EventType;
+import org.apache.ignite.failure.StopNodeFailureHandler;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.DFLT_STORE_DIR;
+
+/**
+ *
```
Comment (alievmirza) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:32]: Please add comment that describes the test

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        DataStorageConfiguration storageCfg = new DataStorageConfiguration();
+
+        storageCfg.getDefaultDataRegionConfiguration()
+            .setPersistenceEnabled(true);
+
+        cfg.setDataStorageConfiguration(storageCfg);
+
+        cfg.setFailureHandler(new StopNodeFailureHandler());
+
+        cfg.setIncludeEventTypes(EventType.EVTS_ALL);
```
Comment (alievmirza) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:69]: The same as previous, do you really need this setting?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+     * Test that tmp checkpoints are getting deleted on startup
+     * @throws Exception
+     */
+    @Test
+    public void testRestartWithTmpCheckpoint() throws Exception {
+        File dbDir = U.resolveWorkDirectory(U.defaultWorkDirectory(), DFLT_STORE_DIR, false);
+
+        String nodeId = UUID.randomUUID().toString();
+
+        File nodeDir = new File(new File(dbDir.getAbsolutePath(),U.maskForFileName(nodeId)), "cp");
```
Comment (alievmirza) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:84]: missed space after comma

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+        nodeDir.mkdirs();
+
+        File tmpFileLeftAfterCrash = new File(nodeDir, "1586870171036-d346c814-aa03-4d66-bf1e-2951a04268f9-END.bin.tmp");
```
Comment (alievmirza) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:88]: Where did this filename come from? Is this just random name?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.events.EventType;
+import org.apache.ignite.failure.StopNodeFailureHandler;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.DFLT_STORE_DIR;
+
```
Comment (alievmirza) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:30]: Remove redundant empty line.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java (hunk 5 above)
Comment (SammyVimes) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/checkpoint/CheckpointTempFilesCleanupOnStartupTest.java:88]: The filename came from the issue's stacktrace and has no real meaning

**Pull request merged**